### PR TITLE
Add tests for sending customer info upon adding listener

### PR DESCRIPTION
**Pull request by tonidero**
This PR adds tests for #576

### CODE REVIEW

File: test/purchases_flutter_test.dart [hunk 1]
```diff
@@ -59,6 +69,72 @@ void main() {
     );
   });
 
+  test('addCustomerInfoUpdateListener does not call listener if does not have one', () async {
```
Comment (NachoSoto) [inline test/purchases_flutter_test.dart:72]:
```suggestion
  test('addCustomerInfoUpdateListener does not call listener if it does not have one', () async {
```

File: test/purchases_flutter_test.dart [hunk 2]
```diff
@@ -5,6 +5,16 @@ import 'package:flutter_test/flutter_test.dart';
 import 'package:purchases_flutter/purchases_flutter.dart';
 
 // ignore_for_file: deprecated_member_use_from_same_package
+
+void _performDartSideChannelMethodCall(String methodName, dynamic params) {
```
Comment (NachoSoto) [inline test/purchases_flutter_test.dart:9]: This is the only part that I'm not no familiar with so I don't want to blindly approve 😇

**Review by NachoSoto — COMMENTED** [review]
The tests look good 👍🏻

**Comment by tonidero** [comment]
@RevenueCat/coresdk bump

File: test/purchases_flutter_test.dart (hunk 2 above)
Comment (aboedo) [reply test/purchases_flutter_test.dart:9]: Could you add a comment here explaining the need and usage of this?

Comment (aboedo) [reply test/purchases_flutter_test.dart:9]: i.e.: 
```dart
/// testing utility that allows you to emit messages as if they were coming directly from the SDK
/// this is useful for ...
/// usage: ...
```

File: test/purchases_flutter_test.dart [hunk 3]
```diff
@@ -59,6 +69,72 @@ void main() {
...
+  test('addCustomerInfoUpdateListener does not call listener if does not have one', () async {
+    Purchases.addCustomerInfoUpdateListener((customerInfo) {
+      fail('listener should not be called if previous customer info was not set');
+    });
+    expect(log, <Matcher>[]);
+  });
+
+  test('addCustomerInfoUpdateListener calls each listener immediately if it has an existing customer info', () async {
+    channel.setMockMethodCallHandler(null);
+    _performDartSideChannelMethodCall('Purchases-CustomerInfoUpdated', mockCustomerInfoResponse);
```
[the remark this answers is not in the document]
Comment (aboedo) [reply test/purchases_flutter_test.dart:81]: let's make that a comment, we'll likely not remember the reasoning a while from now

[the remark this answers is not in the document]
Comment (aboedo) [reply test/purchases_flutter_test.dart:81]: and a maintainer would likely take a while to figure it out

File: test/purchases_flutter_test.dart (hunk 2 above)
Comment (vegaro) [reply test/purchases_flutter_test.dart:14]: oh this is cool, I need it for the setLogHandler

**Review by vegaro — APPROVED** [review]

**Pull request merged**